src/argilla/client/users.py

```diff
@@ -108,14 +109,16 @@ def __init__(
         raise Exception(error_msg)
 
     @property
-    @allowed_for_roles(roles=[UserRole.owner])
-    def workspaces(self) -> Optional[List[WorkspaceModel]]:
+    def workspaces(self) -> Optional[List["WorkspaceModel"]]:
         """Returns the workspace names the current user is linked to.
 
         Returns:
             A list of `WorkspaceModel` the current user is linked to.
         """
-        return users_api_v1.list_user_workspaces(self.__client, self.id).parsed
+        if active_client().user.role == UserRole.owner:
+            return users_api_v1.list_user_workspaces(self.__client, self.id).parsed
+        else:
+            return workspaces_api_v1.list_workspaces_me(self.__client).parsed
```

Comment:
```suggestion
        # TODO: improve this
        connected_user = users_api.whoami_httpx(self.__client).parsed

        if connected_user.role == UserRole.owner:
            return users_api_v1.list_user_workspaces(self.__client, self.id).parsed

        elif connected_user.id == self.id:
            return workspaces_api_v1.list_workspaces_me(self.__client).parsed
        else:
            raise PermissionError(
                f"User with role={connected_user.role} is not allowed to workspaces for user `{self.id}`."
            )
```

Reply:
Or just skip the id check:
```suggestion
        # TODO: improve this
        connected_user = users_api.whoami_httpx(self.__client).parsed

        if connected_user.role == UserRole.owner:
            return users_api_v1.list_user_workspaces(self.__client, self.id).parsed
        return workspaces_api_v1.list_workspaces_me(self.__client).parsed
```

Reply:
We can skip indeed the `user_id` check because any other user than `owner` is not able to use any other method besides `User.me()` initially

Reply:
And regarding the proposal, shouldn't we use the current active user instead of an HTTP request? This way we'll avoid sending 2 requests to Argilla on every call to `User.workspaces`

Reply:
I assume the use-case is changes in the role of the current user, which would be problematic if we're using an `admin` or `annotator`, as we can just use the `User.me` classmethod, and changing the role of the active user will still make the class invalid, as the `role` attribute won't be updated, so IMO there's no need to send that extra HTTP request

Reply:
Yes, it's why I put the TODO to improve this. My concern about your proposal is regarding the use of `active_client` inside this method. This client can be misaligned with the `__client` when the user instance was created, so we can use the wrong info to check the workspaces access